lib/logstash/pluginmanager/list.rb

```diff
@@ -1,38 +1,53 @@
 require 'clamp'
 require 'logstash/namespace'
-require 'logstash/pluginmanager'
 require 'logstash/pluginmanager/util'
 require 'rubygems/spec_fetcher'
 
 class LogStash::PluginManager::List < Clamp::Command
 
-  parameter "[PLUGIN]", "Plugin name to search for, leave empty for all plugins"
+  parameter "[PLUGIN]", "Part of plugin name to search for, leave empty for all plugins"
 
-  option "--group", "NAME", "Show all plugins from a certain group. Can be one of 'output', 'input', 'codec', 'filter'"
+  option "--all", :flag, "Also list plugins installed as dependencies"
+  option "--group", "NAME", "Filter plugins per group: input, output, filter or codec" do |arg|
+    raise(ArgumentError, "should be one of: input, output, filter or codec") unless ['input', 'output', 'filter', 'codec'].include?(arg)
+    arg
+  end
 
   def execute
+    Gem.configuration.verbose = false
+
+    gemfile = LogStash::Gemfile.new(File.new(LogStash::Environment::GEMFILE_PATH, "r+")).load
+
+    specs = all? ? LogStash::PluginManager.find_plugins_gem_specs : LogStash::PluginManager.all_installed_plugins_gem_specs(gemfile)
+    specs = specs.select{|spec| spec.name =~ /#{plugin}/i} if plugin
+    specs = specs.select{|spec| spec.metadata['logstash_group'] == group} if group
```

Comment:
We could probably move this logic into the `pluginmanager` class and call it `list_specs` accepting filter options.